### PR TITLE
feat(i18n): add documentation for custom paths

### PR DESCRIPTION
**Pull request by ematipico**
#### Description (required)

This PR adds documentation for refined locales, a way to customise paths and codes.

I was a bit unsure where to put the new paragraph because these refined locales span different concepts. For this reason, I put it at the end. 

RFC changes: https://github.com/withastro/roadmap/pull/734/commits/dc277b5d02b565f23265269587178cbd18e4a55f

 #### For Astro version: `next release`. See astro PR [#9200](https://github.com/withastro/astro/pull/9200).

### CODE REVIEW

File: src/content/docs/en/guides/internationalization.mdx [hunk 1]
```diff
@@ -282,8 +282,55 @@ Use this like [`getRelativeLocaleUrl`](#getrelativelocaleurl) to return a list o
...
+// These functions are not particularly useful 
+getPathByLocale("fr"); // returns "french"
+getPathByLocale("fr-CA"); // returns "french"
+getPathByLocale("es"); // returns "es"
+getLocaleByPath("french"); // returns "fr" because that's the first code configured
+getLocaleByPath("es"); // returns "es"
+---
+```
+
+If you rely on [the browser language detection](#browser-language-detection), Astro will use the `codes` to determine the `Astro.preferredLocale` and `Astro.preferredLocaleList`. If so, it's advised to write the codes using the same pattern [used by the browser](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept-Language#syntax).
```
Comment (sarah11918) [inline src/content/docs/en/guides/internationalization.mdx:332]:
```suggestion
Astro will use your configured `codes` and [the browser's language detection](#browser-language-detection) to determine `Astro.preferredLocale` and `Astro.preferredLocaleList`. In order to successfully match your visitors' preferences, provide your `codes` using the same pattern [used by the browser](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept-Language#syntax).
```

File: src/content/docs/en/guides/internationalization.mdx [hunk 2]
```diff
@@ -282,8 +282,55 @@ Use this like [`getRelativeLocaleUrl`](#getrelativelocaleurl) to return a list o
...
 
-
 Use this like [`getAbsoluteLocaleUrl`](#getabsolutelocaleurl) to return a list of absolute paths for all the locales.
 
+### `getPathByLocale(locale: string): string`
+
+A function that return the `path` associated to a locale (defined as code). It's particularly useful in case you use [refined locales](#refined-locales).
+
+
+### Refined locales
```
Comment (sarah11918) [inline src/content/docs/en/guides/internationalization.mdx:292]:
```suggestion
### Custom locale paths
```
Initial thinking! We can see what we like here.

**Review by sarah11918 — COMMENTED** [review]
Thanks for these great docs, @ematipico ! Here are some initial thoughts from a first pass through them for your consideration!

File: src/content/docs/en/guides/internationalization.mdx [hunk 3]
```diff
@@ -282,8 +282,55 @@ Use this like [`getRelativeLocaleUrl`](#getrelativelocaleurl) to return a list o
...
+
+Astro allows to map an arbitrary number of locales to a particular URL path. The `locales` array can accept an object, where the `path` key is what you'll see in the URL, and `codes` represents the languages mapped to this URL.
```
[the remark this answers is not in the document]
Comment (ematipico) [reply src/content/docs/en/guides/internationalization.mdx:294]: Updated the suggestion to fix a typo

File: src/content/docs/en/guides/internationalization.mdx [hunk 4]
```diff
@@ -289,8 +305,32 @@ Use this like [`getRelativeLocaleUrl`](#getrelativelocaleurl) to return a list o
...
 `getAbsoluteLocaleUrlList(locale: string, options?: GetLocaleOptions): string[]`
 
-
 Use this like [`getAbsoluteLocaleUrl`](#getabsolutelocaleurl) to return a list of absolute paths for all the locales.
 
+### `getPathByLocale(locale: string): string`
+
+A function that returns the `path` associated to one or more `codes` when [custom locale paths](#custom-locale-paths) are configured.
+
+### Generatng URLs from custom paths
```
Comment (sarah11918) [inline src/content/docs/en/guides/internationalization.mdx:314]:
```suggestion
### Generating URLs from custom paths
```

**Comment by sarah11918** [comment]
@ematipico I did some editing here and moved the content into a few different places where appropriate.

Please take a look and see what you think, and make any suggestions for what you'd like to see different!

**Review by Yan-Thomas — APPROVED** [review]
Everything's LGTM!

File: src/content/docs/en/guides/internationalization.mdx [hunk 5]
```diff
@@ -163,39 +155,65 @@ Set this option when all routes will have their `/locale/` prefix in their URL a
...
+In addition to defining your site's supported `locales` as strings (e.g. "en", "pt-br"), Astro also allows you to map an arbitrary number of [browser-recognized language `codes`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept-Language#syntax) to a custom URL `path`. While locales can be strings of any format as long as they correspond to your project folder structure, `codes` must follow the browser's accepted syntax.
 
-Depending on your deploy host, you may discover transformations in URL paths, so check your deployed site to determine the best syntax for your project.
+Pass an object to the `locales` array with a `path` key to define a custom URL prefix, and `codes` to indicate the languages mapped to this URL. In this case, your `/[locale]/` folder name must match exactly the value of the `path` and your URLs will be generated using the `path` value.
+
+This is useful if you support multiple variations of language (e.g. `"fr"`, `"fr-BR"`, and `"fr-CA"`) and you want to have all these locales mapped under the same URL `/fr/`, or even customize it entirely (e.g. `/french/`):
```
Comment (sarah11918) [inline src/content/docs/en/guides/internationalization.mdx:170]:
```suggestion
This is useful if you support multiple variations of a language (e.g. `"fr"`, `"fr-BR"`, and `"fr-CA"`) and you want to have all these locales mapped under the same URL `/fr/`, or even customize it entirely (e.g. `/french/`):
```

File: src/content/docs/en/guides/internationalization.mdx [hunk 6]
```diff
@@ -289,8 +305,32 @@ Use this like [`getRelativeLocaleUrl`](#getrelativelocaleurl) to return a list o
...
-
 Use this like [`getAbsoluteLocaleUrl`](#getabsolutelocaleurl) to return a list of absolute paths for all the locales.
 
+### `getPathByLocale(locale: string): string`
+
+A function that returns the `path` associated to one or more `codes` when [custom locale paths](#custom-locale-paths) are configured.
+
+### Generating URLs from custom paths
+
+When a `locale` is a custom path, use the `path` as the value for `locale`.
```
Comment (sarah11918) [inline src/content/docs/en/guides/internationalization.mdx:316]:
``````suggestion
When a `locale` is a custom path, use the `path` as the value for `locale`.

For example, use `french` in the i18n helper functions for the `locales` configuration below:

```js
    locales: ["es", "en", {
        path: "french",
        codes: ["fr", "fr-BR", "fr-CA"]
    }],
```
``````

**Comment by sarah11918** [comment]
Merging because these links will work once #5526 is merged!

**Review by ElianCodes — APPROVED** [review]

**Pull request merged**